src/windows/volume.cc

```diff
@@ -86,38 +86,26 @@ HRESULT drivelist::volume::IsVolumeWritable(const wchar_t letter, BOOL *out) {
 }
 
 HRESULT drivelist::volume::HasFileSystem(const wchar_t letter, BOOL *out) {
-  HANDLE handle = drivelist::volume::OpenHandle(letter, 0);
-  if (handle == INVALID_HANDLE_VALUE)
-    return E_HANDLE;
-
-  PARTITION_INFORMATION_EX information;
-  DWORD bytesReturned;
-
-  if (!DeviceIoControl(handle, IOCTL_DISK_GET_PARTITION_INFO_EX, NULL, 0,
-                       &information, sizeof(information),
-                       &bytesReturned, NULL)) {
-    CloseHandle(handle);
-    return E_FAIL;
+  TCHAR drivePath[kVolumePathShortLength];
+  sprintf_s(drivePath, "%c:\\", letter);
+  BOOL result = GetVolumeInformation(drivePath, NULL, 0, NULL,
+                                     NULL, NULL, NULL, 0);
+  if (result) {
+    *out = TRUE;
+    return S_OK;
   }
 
-  // See https://msdn.microsoft.com/en-us/library/windows/desktop/aa365448(v=vs.85).aspx
-  switch (information.PartitionStyle) {
```

Comment:
It might still be worth checking the partition-info, before checking the filesystem with GetVolumeInformation? Since I guess if the partition info is invalid, we know straight away that the filesystem will be invalid too?